SU2_CFD/src/numerics/NEMO/convection/ausm.cpp

```diff
@@ -0,0 +1,425 @@
+/*!
+ * \file NEMO_slau.hpp
+ * \brief Declaration of numerics classes for the NEMO family of schemes,
+ *        including SLAU. The implementation is in NEMO.cpp.
+ * \author F. Palacios, T. Economon
+ * \version 7.0.3 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2020, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#include "../../../../include/numerics/NEMO/convection/ausm.hpp"
+
+CUpwAUSM_NEMO::CUpwAUSM_NEMO(unsigned short val_nDim, unsigned short val_nVar, 
+                             unsigned short val_nPrimVar,
+                             unsigned short val_nPrimVarGrad, 
+                             CConfig *config) : CNumerics(val_nDim, val_nVar,
+                                                          config) {
+
+  /*--- Read configuration parameters ---*/
+  implicit   = (config->GetKind_TimeIntScheme_NEMO() == EULER_IMPLICIT);
+  ionization = config->GetIonization();
+
+  /*--- Define useful constants ---*/
+  nVar     = val_nVar;
+  nPrimVar = val_nPrimVar;
+  nPrimVarGrad = val_nPrimVarGrad;
+  nDim     = val_nDim;
+  nSpecies = config->GetnSpecies();
+
+  FcL    = new su2double [nVar];
+  FcR    = new su2double [nVar];
+  dmLP   = new su2double [nVar];
+  dmRM   = new su2double [nVar];
+  dpLP   = new su2double [nVar];
+  dpRM   = new su2double [nVar];
+  daL    = new su2double [nVar];
+  daR    = new su2double [nVar];
+  rhos_i = new su2double [nSpecies];
+  rhos_j = new su2double [nSpecies];
+  u_i    = new su2double [nDim];
+  u_j    = new su2double [nDim];
+
+  variable = new CNEMOEulerVariable(1, nDim, nVar, nPrimVar, nPrimVarGrad, config);
+
+  RHOS_INDEX    = variable->GetRhosIndex()    ;
+  RHO_INDEX     = variable->GetRhoIndex()     ;
+  P_INDEX       = variable->GetPIndex()       ;
+  T_INDEX       = variable->GetTIndex()       ;
+  TVE_INDEX     = variable->GetTveIndex()     ;
+  VEL_INDEX     = variable->GetVelIndex()     ;
+  H_INDEX       = variable->GetHIndex()       ;
+  A_INDEX       = variable->GetAIndex()       ;
+  RHOCVTR_INDEX = variable->GetRhoCvtrIndex() ;
+  RHOCVVE_INDEX = variable->GetRhoCvveIndex() ;
```

Comment:
If this is the only function of having a NEMOEulerVariable here then either:
Make a static function to return all these indices so that you don't need to instantiate the variable.
Alternatively, and preferably, move the index logic out of the variable and re-used it in both the variable and the numerics, you can for example have a struct with only const members, which would be safe and much less verbose.